### PR TITLE
Navigate and scroll to an extract from a search

### PR DESCRIPTION
**Pull request by paolina92**

### CODE REVIEW

File: assembl/static2/js/app/utils/hashLinkScroll.js [hunk 1]
```diff
@@ -15,7 +16,7 @@ const getContentAreaOffset = () => {
   return offset;
 };
 
-export const scrollToPost = (postElement, smooth = true) => {
+export const scrollToPost = (postElement: any, smooth: boolean = true) => {
```
Comment (MilanBarande) [inline assembl/static2/js/app/utils/hashLinkScroll.js]: I think postElement must be a React.Node

File: assembl/static2/js/app/components/harvesting/harvestingBox.jsx [hunk 2]
```diff
@@ -82,10 +83,13 @@ class DumbHarvestingBox extends React.Component<Props, State> {
     super(props);
     const { extracts, cancelHarvesting } = this.props;
     const hasExtract = extracts ? extracts.length > 0 : false;
-    const extract = this.getCurrentExtract(0);
+    const { hash } = window.location;
+    const hashExtractId = hash !== '' ? hash.split('#')[2] : null;
```
Comment (vincentfretin) [inline assembl/static2/js/app/components/harvesting/harvestingBox.jsx]: hashExtractId can be undefined here if you don't have a second '#" in the url.

File: assembl/static2/js/app/components/harvesting/harvestingMenu.jsx [hunk 3]
```diff
@@ -27,23 +27,29 @@ type State = {
...
-    super(props);
-    this.state = {
-      displayExtractsBox: false
-    };
+  static getDerivedStateFromProps() {
+    const { hash } = window.location;
+    if (hash !== '') {
+      const hashExtractId = hash.split('#')[2];
+      if (hashExtractId) {
+        return { displayExtractsBox: true };
```
Comment (vincentfretin) [inline assembl/static2/js/app/components/harvesting/harvestingMenu.jsx]: You want to display extracts box only if hashExtractId is in extracts, not expand all box on all posts.

File: assembl/static2/js/app/utils/hashLinkScroll.js (hunk 1 above)
Comment (vincentfretin) [inline assembl/static2/js/app/utils/hashLinkScroll.js]: Please don't use `any`

File: assembl/static2/js/app/components/harvesting/harvestingBox.jsx [hunk 4]
```diff
@@ -82,10 +83,13 @@ class DumbHarvestingBox extends React.Component<Props, State> {
...
+    const extractIndex = hashExtractId && extracts ? extracts.indexOf(this.getCurrentExtractById(hashExtractId)) : 0;
```
Comment (vincentfretin) [inline assembl/static2/js/app/components/harvesting/harvestingBox.jsx]: `this.getCurrentExtractById(hashExtractId)` may return null if hashExtractId is not found.

Comment (vincentfretin) [reply assembl/static2/js/app/components/harvesting/harvestingBox.jsx]: [].indexOf(null) returns -1
you may set extractIndex to 0 if you get a -1

File: assembl/static2/js/app/components/harvesting/harvestingMenu.jsx (hunk 3 above)
Comment (paolina92) [reply assembl/static2/js/app/components/harvesting/harvestingMenu.jsx]: yes good catch!

File: assembl/static2/js/app/components/harvesting/harvestingBox.jsx (hunk 2 above)
Comment (paolina92) [reply assembl/static2/js/app/components/harvesting/harvestingBox.jsx]: I handle the case when hashExtractId is undefined after

File: assembl/static2/js/app/components/harvesting/harvestingBox.jsx (hunk 4 above)
Comment (paolina92) [reply assembl/static2/js/app/components/harvesting/harvestingBox.jsx]: done

File: assembl/static2/js/app/utils/hashLinkScroll.js (hunk 1 above)
Comment (vincentfretin) [reply assembl/static2/js/app/utils/hashLinkScroll.js]: @paolina92 you forgot this comment.

File: assembl/static2/js/app/components/search.jsx [hunk 5]
```diff
@@ -86,6 +87,14 @@ function getPostUrl(ideaId, postId, phaseIdentifier, slug) {
       questionIndex: 1,
       element: postBase64id
     });
+  } else if (phaseIdentifier === 'thread' && extractId) {
```
Comment (vincentfretin) [inline assembl/static2/js/app/components/search.jsx]: You shouldn't have `phaseIdentifier === 'thread'` here.

Comment (amensouissi) [reply assembl/static2/js/app/components/search.jsx]: PHASES.thread

**Pull request merged**